BuildGenerated.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+set -e
+
+# NuGet package cache directory for restore/build.
+NUGET_PACKAGE_DIR=$(pwd)/Packages
+# Final output directory of NuPkgs.
+NUPKG_DIR=$(pwd)/NuPkgs/Generated
+# Build configuration to build/pack.
+BUILD_CONFIGURATION=Release
+# Directory in which to download discovery docs.
+DISCOVERY_DOC_DIR=$(pwd)/DiscoveryJson
+# Code generation directory
+CODE_GENERATION_DIR=$(pwd)/Src/Generated
+# Directory containing tools used during the build.
+TOOLS_DIR=$(pwd)/Src/Tools
+
+export NUGET_PACKAGES=$NUGET_PACKAGE_DIR
+
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    --skipdownload)
+      SKIPDOWNLOAD=TRUE
+      ;;
+    --skipgenerate)
+      SKIPDOWNLOAD=TRUE
+      SKIPGENERATE=TRUE
+      ;;
+    --skipbuild)
+      SKIPDOWNLOAD=TRUE
+      SKIPGENERATE=TRUE
+      SKIPBUILD=TRUE
+      ;;
+    --skippack)
+      SKIPDOWNLOAD=TRUE
+      SKIPGENERATE=TRUE
+      SKIPBUILD=TRUE
+	  SKIPPACK=TRUE
+      ;;
+    --onlydownload)
+      SKIPGENERATE=TRUE
+      SKIPBUILD=TRUE
+	  SKIPPACK=TRUE
+      ;;
+    --onlygenerate)
+      SKIPDOWNLOAD=TRUE
+      SKIPBUILD=TRUE
+	  SKIPPACK=TRUE
+      ;;
+    --onlybuild)
+      SKIPDOWNLOAD=TRUE
+      SKIPGENERATE=TRUE
+	  SKIPPACK=TRUE
+      ;;
+    --onlypack)
+      SKIPDOWNLOAD=TRUE
+      SKIPGENERATE=TRUE
+	  SKIPBUILD=TRUE
+      ;;
+    *)
+      echo ERROR: Invalid argument to BuildGenerated.sh: \'$key\'
+	  exit 1
+  esac
+  shift
+done
+
+if [ -z ${SKIPDOWNLOAD+x} ]; then
+  # Delete all discovery docs. Ignore failure if this dir doesn't exist.
```

Comment:
Any reason not to just use `rm -rf`, which doesn't fail if the directory doesn't exist? Admittedly we then don't get the same diagnostic messages, but I don't think it's a massive deal.